docs/autoprocess.py

```diff
@@ -0,0 +1,401 @@
+        categories = set()
+        for process in process_list:
+            categories.add(process.get('category', 'uncategorized'))
+        return list(categories)
+
+    def make_field(self, field_name, field_body):
+        """
+        Handy function for filling content into nodes.field
+
+        :param string field_name: Field name of the field
+        :param field_name: Field body if the field
+        :type field_name: str or instance of docutils.nodes
+        :rtype: nodes.field
+        """
+        field = nodes.field()
+
+        name = nodes.field_name()
+        name += nodes.Text(field_name)
+
+        body = nodes.field_body()
+        paragraph = nodes.paragraph()
+        if isinstance(field_body, six.string_types):
+            # This is the case when field_body is just a string:
+            paragraph += nodes.Text(field_body)
+        else:
+            # This is the case when field_body is a complex node:
+            # useful when constructing nested field lists
+            paragraph += field_body
+
+        body += paragraph
+        field += name
+        field += body
+        return field
+
+    def make_properties_list(self, data):
+        """
+        Fill the `data` into a properties list and return it.
+
+        :param dict data: the content of the property list to make
+        :rtype: nodes.field_list
+        """
+        # Make a copy so the function parameter is not modified.
+        data = data.copy()
+
+        properties_list = nodes.field_list()
+
+        # Remove the name (it is already used in field_name...)
+        data.pop('name')
+
+        # changing the order of elements in this list affects
+        # the order in which they are displayed
+        property_names = [
+            'label',
+            'type',
+            'description',
+            'required',
+            'disabled',
+            'hidden',
+            'default',
+            'placeholder',
+            'validate_regex',
+            'choices',
+            'collapse',
+            'group']
+
+        for name in property_names:
+            if name in data:
+                value = data.pop(name)
+
+                # Value should be formatted in code-style (=literal) mode
+                if name in ['type', 'default', 'placeholder', 'validate_regex']:
+                    literal_node = nodes.literal(str(value), str(value))
+                    properties_list += self.make_field(name, literal_node)
+
+                # Special formating of `value` is needed if name == 'choices'
+                elif name == 'choices':
+                    bullet_list = nodes.bullet_list()
+                    for choice in value:
+                        list_item = nodes.list_item()
+                        paragraph = nodes.paragraph()
+                        label = nodes.Text(choice['label'] + ': ')
+                        val = nodes.literal(choice['value'], choice['value'])
+
+                        paragraph += label
```

Comment:
Move `paragraph` definition just before it is used. Or maybe even `paragraph = nodes.paragraph() + label + val`?